### PR TITLE
UCP/AM: Aligned function names with files hierarchy.

### PR DESCRIPTION
**Pull request by rakhmets**
## What
Aligned function names with files hierarchy.

+ removed duplicate function filling zcopy footer.

## Why ?
Fixed https://github.com/openucx/ucx/pull/8019#discussion_r825461677

### CODE REVIEW

File: src/ucp/am/eager.inl [hunk 1]
```diff
@@ -16,21 +16,22 @@
...
+static UCS_F_ALWAYS_INLINE size_t
+ucp_am_eager_send_req_total_size(ucp_request_t *req)
 {
     return req->send.state.dt_iter.length +
            req->send.msg_proto.am.header_length;
 }
 
 static UCS_F_ALWAYS_INLINE void
-ucp_am_pack_user_header(void *buffer, ucp_request_t *req)
+ucp_am_eager_pack_user_header(void *buffer, ucp_request_t *req)
```
Comment (brminich) [inline src/ucp/am/eager.inl:34]: can these funcs be used for rndv later? Maybe no need to rename them

Comment (rakhmets) [reply src/ucp/am/eager.inl:34]: yes, it will be re-used by rndv. i'll revert this rename.

Comment (rakhmets) [reply src/ucp/am/eager.inl]: Reverted renaming of `ucp_am_fill_header` and `ucp_am_pack_user_header`.

File: src/ucp/am/eager.inl [hunk 2]
```diff
@@ -23,7 +23,8 @@ ucp_am_fill_header(ucp_am_hdr_t *hdr, ucp_request_t *req)
     hdr->header_length = req->send.msg_proto.am.header_length;
 }
 
-static UCS_F_ALWAYS_INLINE size_t ucp_am_send_req_total_size(ucp_request_t *req)
+static UCS_F_ALWAYS_INLINE size_t
+ucp_am_eager_send_req_total_size(ucp_request_t *req)
```
Comment (brminich) [inline src/ucp/am/eager.inl:27]: i think this is also common for all protocols

**Review by brminich — DISMISSED** [review]

**Review by yosefe — DISMISSED** [review]

**Review by yosefe — APPROVED** [review]

**Review by brminich — APPROVED** [review]

**Pull request merged**